zipkin-common/src/main/scala/com/twitter/zipkin/storage/SpanStore.scala

```diff
@@ -149,6 +149,23 @@ class InMemorySpanStore extends SpanStore with CollectAnnotationQueries {
       .toList
   }
 
+  override protected def getTraceIdsByDuration(
+    serviceName: String,
```

Comment:
Thanks for looking deep enough to have questions like this!

> Q1 - if I am reading sql query correctly, the duration limit applies to
individual span duration, not the overall trace duration, is that correct?

Since the query only applies to root spans, it is essentially the same as
trace. There's no trace table in SQL as a trace is just where id = traceid

> Q2 - what are the semantics of serviceName here? Any service involved in
a span satisfying the duration limit? What about services involved in a
long trace, but not in the long span in that trace?
So this doesn't constrain the search to only service names in the root
span. It applies to any service in the trace, at the moment (I think).
Probably we should make a test about this nuance. What's your goal? Limit
to service names only in the root span?

Keep in mind that other queries act similarly, ex annotation query isn't
bound to only annotations from that service iirc. In other words, this may
be a bigger issue.

> Q3 - shouldn't we include span name in the search?
You are only seeing a partial query. The scala lib does this thing called
query slice. I don't like that but don't have time to rewrite all the
backends. This is part of a deprecated class.

Ex in the java port, you'd see this implemented differently (still passing
tests). My hope is that we delete anorm once the java port has a release.

Cassandra should write directly to SpanStore as opposed to via the
deprecated SliceQuery approach. Would love your help with that.